couler/core/syntax/loop.py

```diff
@@ -19,22 +19,33 @@
 from couler.core.templates import Step, output
 
 
-def map(function, input_list):
+def map(function, *other):
     """
     map operation of Couler
     """
+
     # Enforce the function to run and lock to add into step
+    # Checking correct syntax
     if callable(function):
         states._update_steps_lock = False
-        # TODO (terrytangyuan): Support functions with multiple arguments.
-        para = input_list[0]
-        inner = function(para)
+
+        para = []
+        x = 0
+
+        while x < len(other):
+            para.append(other[x][0])
+            x += 1
+
+        inner = function(*para)
+
         if inner is None:
             raise SyntaxError("require function return value")
         states._update_steps_lock = True
+
     else:
         raise TypeError("require loop over a function to run")
 
+    # creates step in .yaml form and adds parameters
```

Comment:
Use complete sentences

Reply:
I deleted this comment since I'm not confident enough to explain in detail someone else's code that I do not completely understand.